feat: add record_exception in the datadog api (#4567)
DataDog/dd-trace-rb: docs/GettingStarted.md

You can tag additional information to the current active span from any method. Note however that if the method is called and there is no span currently active `active_span` will be nil.

```ruby
# Example:  adding tag to active span

current_span = Datadog::Tracing.active_span
current_span.set_tag('my_tag', 'my_value') unless current_span.nil?
```

You can record an exception in the current span. It adds the recorded exception as a span event.
```ruby
# Example: recording an exception in the active span
rescue => e

>   current_span = Datadog::Tracing.active_span

Reviewer: Prefer safe navigator operation to nilcheck

>   if !current_span.nil?

Author: ```suggestion
  if current_span
```

>   if current_span
>     current_span.record_exception(e, attributes: { foo: "bar" })
>   end

Author: ```suggestion
  current_span?.record_exception(e, attributes: { foo: "bar" })
```

current_span.set_tag('my_tag', 'my_value') unless current_span.nil?
```

> You can record an exception in the current span. It will add the recorded exception as a span event. If you also want to set the span error tags, you should set `escaped` to true.

Reviewer: The example code is broken due to 

1. It is invoking with positional argument, not keyword arguments
2. Does not handle correctly when `Datadog::Tracing.active_span` returns `nil`
3. `false` instead of `False` (why displaying a method default which could be ignored?)

Author: Yes, it is an opentelemetry term. It is implemented in Python for unhandled exceptions.
TL;DR: it is not in ruby so I'll just remove it

> You can record an exception in the current span. It will add the recorded exception as a span event. If you also want to set the span error tags, you should set `escaped` to true.

Reviewer 2: ```suggestion
You can record an exception in the current span. It adds the recorded exception as a span event. If you also want to set the span error tags, set `escaped` to true.
```

```ruby

> # e.g. recording an exception in the active span

Reviewer 2: ```suggestion
# Example recording an exception in the active span
```
